client/src/components/pub-search.jsx

```diff
@@ -0,0 +1,158 @@
+          </Link>
+          <h4>Search for Publication (by title or Twitter handle)</h4>
+          <div className="container-fluid">
+            <div className="container">
+              <div className="FormCenter">
+                <form className="FormField">
+                  <div className="FormField">
+                    <label for="searchTitle">Publication Title</label>
+                    <input
+                      type="text"
+                      className="form-control"
+                      id="searchTitle"
+                      placeholder="Enter name of publication"
+                      name="searchTitle"
+                      value={this.state.searchTitle}
+                      onChange={this.handleChange}
+                    />
+                  </div>
+                  <div className="form-group">
+                    <label for="searchTwitterHandle">Twitter Handle</label>
+                    <input
+                      type="text"
+                      className="form-control"
+                      id="searchTwitterHandle"
+                      placeholder="Enter Twitter Handle"
+                      name="searchTwitterHandle"
+                      value={this.state.searchTwitterHandle}
+                      onChange={this.handleChange}
+                    />
+                  </div>
+                  <br></br>
+                  <button className="btn btn-sm btn-secondary searchSubmit" onClick={this.handleSubmit}>
+                    Submit{' '}
+                  </button>
+                  <button className="btn btn-sm btn-secondary searchCancel" onClick={this.handleCancel}>
+                    Cancel{' '}
+                  </button>
+                </form>
+              </div>
+            </div>
+          </div>
+        </div>
+      );
+    } else {
+      return (
+        <div>
+          <Link to="/pub-entry" className="btn btn-link text-secondary">
+            <span className="text-secondary">new publication</span>
+          </Link>
+          <Link to="/pub-search" className="btn btn-link text-secondary">
+            <span className="text-secondary">new search</span>
+          </Link>
+          <table className="table">
+            <thead>
+              <tr>
+                <th scope="col">Publication</th>
+                <th scope="col">Twitter Handle</th>
+                <th scope="col">{'  '}</th>
+              </tr>
+            </thead>
+            <tbody>
+              {this.state.data.map((results) => (
+                <tr>
+                  <th scope="row">{results.name}</th>
+                  <td>{results.twitterHandle}</td>
+                  <td>
+                    <button className="btn btn-sm btn secondary add-mypubs" onClick={() => this.addMyPubs()}>
+                      Add to My Pubs
+                    </button>
+                  </td>
+                </tr>
+              ))}
+            </tbody>
+          </table>
```

Comment:
The publications data table is duplicated here from `PubDisplay`. A DRY approach would be to extract this chunk out into a standalone component that can be used by both `PubDisplay` and `PubSearch`.

Reply:
Maybe you can talk me through this. I know it's got to be like the `userProfile` component in that it pulls the data from `props` but I just can't wrap my head around it...